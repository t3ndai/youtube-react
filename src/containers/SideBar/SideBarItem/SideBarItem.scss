.sidebar-item {
  background: #ff55ff;
  span {
    i.icon {
      margin-right: 20px;
      color: #888888;
    }
  }

  &.highlight-item {
    span {
      font-weight: 600;
    }

    i.icon {
      color: #ff0002;
    }
  }
}

.sidebar-item-alignment-container {
  display: flex;
  align-items: center;
}
