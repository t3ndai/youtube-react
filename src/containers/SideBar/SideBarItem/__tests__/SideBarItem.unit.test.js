import React from "react";
import { shallow } from "enzyme";
import { SideBarItem } from "../SideBarItem";

describe("SideBarItem", () => {
  test("it renders an empty SideBarItem", () => {
    const wrapper = shallow(<SideBarItem />);
    expect(wrapper).toMatchSnapshot();
  });

  test('it renders SideBarItem with label="trending"', () => {
    const wrapper = shallow(<SideBarItem label="trending" />);
    expect(wrapper).toMatchSnapshot();
  });

  test('it renders a SideBarItem with label="trending" and icon="fire"', () => {
    const wrapper = shallow(<SideBarItem icon="fire" />);
    expect(wrapper).toMatchSnapshot();
  });
});
