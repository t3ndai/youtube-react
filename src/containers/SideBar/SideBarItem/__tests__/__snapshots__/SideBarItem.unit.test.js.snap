// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBarItem it renders SideBarItem with label="trending" 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span>
      trending
    </span>
  </div>
</MenuItem>
`;

exports[`SideBarItem it renders a SideBarItem with label="trending" and icon="fire" 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        name="fire"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;

exports[`SideBarItem it renders an empty SideBarItem 1`] = `
<MenuItem
  className="sidebar-item "
>
  <div
    className="sidebar-item-alignment-container"
  >
    <span>
      <Icon
        as="i"
        size="large"
      />
    </span>
    <span />
  </div>
</MenuItem>
`;
